a3p-integration/README.md

```diff
@@ -1,38 +1,95 @@
+
+The end-to-end integration test relies on the `@agoric/synthetic-chain` test tool (published to NPM from the `agoric-3-proposals` repository), and uses yarn 4 (configured through corepack).
+
+First time setup (or after updates to `@agoric/synthetic-chain`):
+```sh
+corepack enable
+yarn install
 ```
+
+### Generate a docker image with the `agoric-sdk` chain software
+
+The first proposal contained in this end-to-end integration test performs an upgrade of the agoric-3 synthetic chain to an `UNRELEASED_UPGRADE` plan name (or the corresponding upgrade plan name for release branches). It loads the docker image `ghcr.io/agoric/agoric-sdk:unreleased` for the software implementing that upgrade (both in the `master` branch or in release branches). These values are configured in the `package.json` found in the `proposals/a:upgrade-next` folder.
+
+The upgrade handler is implemented by the code in the enclosing `agoric-sdk` repository. After any change to the chain software or vat code upgraded through core proposals, the image must be regenerated.
+
+After code changes in `agoric-sdk`
+```sh
+make -C ../packages/deployment docker-build-sdk
+```
+
+## Run the integration test
+
+The `@agoric/synthetic-chain` test runner can be invoked through yarn.
+
+To build the agoric-3 synthetic chain images with the unreleased chain software upgrade:
+```sh
 yarn synthetic-chain build
+```
+
+To run the unit tests:
+```sh
 yarn synthetic-chain test
+# or for an interactive session
 yarn synthetic-chain test --debug
 ```
 
-# Package management
+# Proposal structure
+
+## Layering
+
+Proposal packages are under the `a3p-integration/proposals` folder, and are separate from each other, from the `a3p-integration` test environment, and from the enclosing `agoric-sdk` repository. They run inside the docker image, and cannot access SDK code. The are executed in lexical order.
+
+The only proposal in this end-to-end `a3p-integration` test is named `a:upgrade-next` and performs a chain software upgrade proposal.
+
+The details of a proposal are configured through the `agoricProposal` field of its `package.json`. For a chain software upgrade, the `type` is `"Software Upgrade Proposal"`, and the relevant fields are `sdkImageTag`, `planName` and `upgradeInfo`. The first, `sdkImageTag`, is the docker image tag to use that contains the upgraded chain software, and has a value of `unreleased`, which is expected to have been built from the enclosing `agoric-sdk` repository. The other 2 are details of the proposed chain software upgrade. The `planName` is `UNRELEASED_UPGRADE` in the `master` branch, or `agoric-upgrade-*` in release branches. The `upgradeInfo` can be configured to include a `coreProposal` field to run any core proposals not otherwise configured in the chain software's upgrade handler (see `CoreProposalSteps` in `golang/cosmos/app/app.go`).
```

Comment:
I wanted this to document the integration with the tool, but did extend on how the tool works since the doc is lacking there, and couldn't make references.

Reply:
To not slow this down, I'm fine with accepting refactoring to there as future work